lib/api/actions/functions.php

```diff
@@ -91,130 +91,122 @@ function beans_add_smart_action( $hook, $callback, $priority = 10, $args = 1 ) {
- *                           Use NULL to keep the original value.
- * @param callback $callback Optional. The name of the new function you wish to be called.
- *                           Use NULL to keep the original value.
- * @param int      $priority Optional. The new priority.
- *                           Use NULL to keep the original value.
- * @param int      $args     Optional. The new number of arguments the function accept.
- *                           Use NULL to keep the original value.
- *
- * @return bool Will always return true.
+ * @since 1.5.0 Made WPCS compliant and improved action parameter filtering.
+ *
+ * @param string        $id       The action's Beans ID, a unique ID tracked within Beans for this action.
+ * @param string|null   $hook     Optional. The new action's event name to which the $callback is hooked.
+ *                                Use NULL to keep the original value.
+ * @param callable|null $callback Optional. The new callback (function or method) you wish to be called.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $priority Optional. The new priority.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $args     Optional. The new number of arguments the $callback accepts.
+ *                                Use NULL to keep the original value.
+ *
+ * @return bool
  */
 function beans_modify_action( $id, $hook = null, $callback = null, $priority = null, $args = null ) {
-
-	// Remove action.
-	if ( $current = _beans_get_current_action( $id ) ) {
-		remove_action( $current['hook'], $current['callback'], $current['priority'], $current['args'] );
+	$action = _beans_build_action_for_valid_args( $hook, $callback, $priority, $args );
+	// If no changes were passed in, there's nothing to modify. Bail out.
+	if ( empty( $action ) ) {
+		return false;
 	}
 
-	$action = array_filter( array(
-		'hook'     => $hook,
-		'callback' => $callback,
-		'priority' => $priority,
-		'args'     => $args,
-	) );
-
-	// Merge modified.
-	$action = _beans_merge_action( $id, $action, 'modified' );
+	$current_action     = _beans_get_current_action( $id );
+	$has_current_action = ! empty( $current_action ) && is_array( $current_action );
 
-	// Replace if needed.
-	if ( $current ) {
-
-		$action = array_merge( $current, $action );
+	// If the action is registered, let's remove it.
+	if ( $has_current_action ) {
+		remove_action( $current_action['hook'], $current_action['callback'], $current_action['priority'], $current_action['args'] );
+	}
 
-		add_action( $action['hook'], $action['callback'], $action['priority'], $action['args'] );
+	// Merge the modified parameters and register with Beans.
+	$action = _beans_merge_action( $id, $action, 'modified' );
 
+	// If there is no action to modify, bail out.
+	if ( ! $has_current_action ) {
+		return false;
```

Comment:
Instead of returning `true` here, let's flip it and bail out.  There is no current action and nothing to register with WordPress.  `false` lets us know that we did not modify anything.